re-enable old unit tests, add new boms, new boosters and new unit tests

ajm01 opened the pull request:
Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

mattbsox commented on proposals/boost/boost-maven/boost-boms/boost-mp21-apis-bom/pom.xml:
```diff
@@ -0,0 +1,119 @@
+	            <artifactId>microprofile-opentracing-api</artifactId>
+	            <version>${opentracing-version}</version>
+	        </dependency>
+	        <dependency>
```
Should the boosters be included in the API BOM?

ajm01 replied:
i don't think so - the booster dependencies are listed in the individual framework boms, which also import the api-boms 

ie: booster-ee8-bom lists all the individual booster dependencies for ee8 (at the correct versions) and imports the booster-ee8-apis-bom - i think that is for satisfying and separating both compile only targets and the boost package runtime target.

chyt replied:
I'm a bit confused. I think Matt is saying that the boosters should not be included in the API boms, whereas in the line that is referenced they are. I agree that they should be removed.

ajm01 replied:
Oh, ok, I see what you are saying - i am not sure how that happened, might have been a cut and paste issue. I will remove them.

ajmau01 committed:
re-enable old unit tests, add new boms, new boosters and new unit tests

Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

remove booster dependencies from the api boms based on matt's comments

Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

fix version numbers and related unit tests based on scott's review comment

Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

add copyrights and fix comments based on scott's review comment

Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

fix boms to use new booster version numbers

Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

fixed missed update for jsonp version in the ee8 bom

Signed-off-by: Andrew J. Mauer <ajmauer@gmail.com>

scottkurz commented on proposals/boost/boost-common/src/main/java/boost/common/config/ConfigConstants.java:
```diff
@@ -100,12 +100,19 @@
     public static final String JDBC_43 = "jdbc-4.3";
     public static final String JPA_21 = "jpa-2.1";
     public static final String JPA_22 = "jpa-2.2";
+    public static final String CDI_12 = "cdi-1.2";
```
Wondering if we should do more to separate out Liberty, TomEE, etc. - related strings...not that this PR started this.

scottkurz commented on proposals/boost/boost-maven/boost-boms/boost-ee7-apis-bom/pom.xml:
```diff
@@ -36,6 +36,16 @@
                 <artifactId>javax.transaction-api</artifactId>
                 <version>1.2</version>
             </dependency>
+            <dependency>
```
indentation is a bit off

scottkurz commented on proposals/boost/boost-maven/boost-boms/boost-ee8-apis-bom/pom.xml:
```diff
@@ -46,6 +46,11 @@
+				<groupId>javax.json.bind</groupId>
+				<artifactId>javax.json.bind-api</artifactId>
+				<version>1.0</version>
+			</dependency>
```
Just a thought:  did we ever consider pulling in the EE7 / EE8 aggregate dependencies?

scottkurz replied:
Same would apply to MP BOMs.

scottkurz commented on proposals/boost/boost-maven/boost-boms/boost-mp20-apis-bom/pom.xml:
```diff
@@ -53,6 +53,11 @@
+	            <groupId>javax.json.bind</groupId>
+	            <artifactId>javax.json.bind-api</artifactId>
+	            <version>${jsonp-version}</version>
+	        </dependency>
```
This is supposed to be `jsonp-version` not `jsonb-version`?

scottkurz commented on proposals/boost/boost-maven/boost-boms/booster-ee7-bom/pom.xml:
```diff
@@ -43,7 +43,7 @@
 				<!-- Probably will be 1.0 for initial GA -->
-				<version>0.1-SNAPSHOT</version>
+				<version>0.2.0-SNAPSHOT</version>
 				<scope>provided</scope>
```
Comment above is now outdated... I think a better way to pattern versioning after the technology version might be to do a 2.0-alpha1, 2.0-beta1, etc.... which both indicates that this is pre-GA but also allows for real releases.

scottkurz commented on proposals/boost/boost-maven/boost-boosters/booster-mpFaultTolerance11/pom.xml:
```diff
@@ -0,0 +1,30 @@
+	<dependencies>
+		<dependency>
+		    <groupId>org.eclipse.microprofile.fault-tolerance</groupId>
+            <artifactId>microprofile-fault-tolerance-api</artifactId>
```
indentation

scottkurz reviewed (changes requested):
I didn't get to the tests.. the main comment there is on the POM versioning.

scottkurz commented on proposals/boost/boost-maven/boost-runtimes/runtime-openliberty/src/test/java/io/openliberty/boost/runtimes/boosters/MPOpenTracingBoosterTest.java:
```diff
@@ -0,0 +1,76 @@
+package io.openliberty.boost.runtimes.boosters;
```
missing copyright

scottkurz commented on proposals/boost/boost-maven/boost-runtimes/runtime-openliberty/src/test/java/io/openliberty/boost/runtimes/boosters/MPMetricsBoosterTest.java:
```diff
@@ -0,0 +1,53 @@
+
+    /**
+     * Test that the mpConfig-1.3 feature is added to server.xml when the MPConfig
+     * booster version is set to 0.2-SNAPSHOT
```
these comments are going to become outdated with the new versioning.

scottkurz commented on proposals/boost/boost-maven/boost-runtimes/runtime-openliberty/src/test/java/io/openliberty/boost/runtimes/boosters/MPFaultToleranceBoosterTest.java:
```diff
@@ -0,0 +1,53 @@
+        LibertyServerConfigGenerator serverConfig = new LibertyServerConfigGenerator(
+                outputDir.getRoot().getAbsolutePath(), logger);
+
+        LibertyMPFaultToleranceBoosterConfig libMPFTConfig = new LibertyMPFaultToleranceBoosterConfig(BoosterUtil.createDependenciesWithBoosterAndVersion(LibertyMPFaultToleranceBoosterConfig.class, "0.1.1-SNAPSHOT"), logger);
```
Would this "...SNAPSHOT" string need to change during a real release?  If so, we should fix this.

scottkurz commented on proposals/boost/boost-maven/boost-runtimes/runtime-openliberty/src/main/java/boost/runtimes/openliberty/boosters/LibertyMPOpenTracingBoosterConfig.java:
```diff
@@ -29,9 +29,13 @@ public LibertyMPOpenTracingBoosterConfig(Map<String, String> dependencies, Boost
     @Override
     public String getFeature() {
-        if (getVersion().equals(MP_20_VERSION)) {
+        if (getVersion().equals(MP_OPENTRACING_VERSION_11)) {
```
These tests read oddly.. maybe we should change the string returned to "FEATURE_OPENTRACING_11" ?

scottkurz commented on proposals/boost/boost-maven/boost-runtimes/runtime-openliberty/src/test/java/io/openliberty/boost/runtimes/utils/BoosterUtil.java:
```diff
@@ -1,11 +1,11 @@
+import boost.common.boosters.AbstractBoosterConfig;
+import boost.common.boosters.JDBCBoosterConfig;
 
 public class BoosterUtil {
```
I realize you didn't create this class..but it could really use some Javadoc... what is mapping to what here in this map?

scottkurz commented:
Moved relevant comments over to: https://github.com/dev-tools-for-enterprise-java/boost/pull/328   so closing this.
